### ISSUE TITLE
Rounded corners persist with `dwindle.no_gaps_when_only = yes` and only one window open

### ISSUE THREAD

**Issue by Mango0x45** [hyprwm/Hyprland#5887]
### Hyprland Version

<details> <summary>System/Version info</summary>

```sh
Hyprland, built from branch  at commit fe7b748eb668136dd0558b7c8279bfcd7ab4d759  (props: bump version to 0.39.1).
Date: Tue Apr 16 16:01:03 2024
Tag: v0.39.1, commits: 4460

flags: (if any)


System Information:
System name: Linux
Node name: mango-os
Release: 6.8.9-arch1-1
Version: #1 SMP PREEMPT_DYNAMIC Thu, 02 May 2024 17:49:46 +0000


GPU information: 
00:02.0 VGA compatible controller [0300]: Intel Corporation TigerLake-LP GT2 [Iris Xe Graphics] [8086:9a49] (rev 01) (prog-if 00 [VGA controller])


os-release: NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://gitlab.archlinux.org/groups/archlinux/-/issues"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo


plugins:


```

</details>

### Bug or Regression?

Bug

### Description

When I set `dwindle.no_gaps_when_only = yes` in my configuration, I expect to have no gaps when I have only one window open. This works. I also expect that when I only have one window open, that the border and corner rounding are disabled, but this is only sometimes the case.

My configuration can be found in whole [here][1]. (Ignore anything that begins with `__`, those are macros).

You can see here what happens when I open a window on a new empty workspace. This looks fine.

![image](https://github.com/hyprwm/Hyprland/assets/57815710/5a362f1e-3e6d-4fa3-87d7-eca12aa226c2)

This is what it looks like when I open a second window. This is also fine.

![image](https://github.com/hyprwm/Hyprland/assets/57815710/57342b7f-66b4-4f22-96df-ba3a0d6968aa)

This is what it looks like when I close the second window. This is not fine.

![image](https://github.com/hyprwm/Hyprland/assets/57815710/8889dacf-e7c8-4448-9d2d-a0986bf98876)

[1]: https://paste.thomasvoss.com/444

### How to reproduce

Add the following to your Hyprland configuration:

```
dwindle {
	no_gaps_when_only = yes
}
```

After that move to a new workspace and open two windows. Kill one of those two windows.

**Comment by sungyoonc** [comment]
It should be fixed in #5552 and the fix should be already included in v0.40.0.

**Comment by vaxerski** [comment]
dupe
